src/Concordium/Client/Types/Contract/Parameter.hs

```diff
@@ -44,6 +44,10 @@ import GHC.Integer (remInteger, modInteger)
 encodeParameter :: SchemaType -> AE.Value -> Either String ByteString
 encodeParameter typ params = S.runPut <$> putJSONUsingSchema typ params
 
+-- |Deserialize binary to JSON using `SchemaType` or fail with an error message.
+decodeParameter :: SchemaType -> ByteString -> Either String AE.Value
+decodeParameter typ = S.runGet $ getJSONUsingSchema typ
```

Comment:
The name of this is rather odd, since it has nothing to do with parameters. It is used for events as well.

Reply:
Also, for decoding events it is **essentiall** that the entire event is consumed while parsing. Otherwise we could easily have buggy, misleading prints.

So we have to make sure that `getJSONUsingSchema`  consumes the entire input.

Reply:
I agree. The same goes for the already existing `encodeParameter` as well, but I wanted to follow the style of the current naming. 

This seems to be an issue of more general nature, so moving forward should I a) rename oddly named functions or b) rather name the ones I add suitably, even though there may be related functions with odd names?

I'll add the end-of-input check for getJSONUsingSchema 👍🏼

Reply:
Yes, this is a legacy from the time when we only had schemas for parameters.
I think we should rename them (and the file), but perhaps that should be done in a separate PR.